### PR TITLE
Make ingest external file backward compatible

### PR DESCRIPTION
**Pull request by shuzhang1989**

### CODE REVIEW

File: db/external_sst_file_ingestion_job.cc [hunk 1]
```diff
@@ -291,24 +291,26 @@ Status ExternalSstFileIngestionJob::GetIngestedFileInfo(
   file_to_ingest->version = DecodeFixed32(version_iter->second.c_str());
 
   auto seqno_iter = uprops.find(ExternalSstFilePropertyNames::kGlobalSeqno);
-  if (file_to_ingest->version == 2) {
```
Comment (IslamAbdelRahman) [inline db/external_sst_file_ingestion_job.cc]: we should keep all these checks for V2, but add a special condition for V1.

File: db/external_sst_file_ingestion_job.cc [hunk 2]
```diff
@@ -291,24 +291,26 @@ Status ExternalSstFileIngestionJob::GetIngestedFileInfo(
...
+  // Set the global sequence number
+  file_to_ingest->original_seqno = DecodeFixed64(seqno_iter->second.c_str());
+  file_to_ingest->global_seqno_offset = props->properties_offsets.at(
+      ExternalSstFilePropertyNames::kGlobalSeqno);
 
-    if (file_to_ingest->global_seqno_offset == 0) {
-      return Status::Corruption("Was not able to find file global seqno field");
-    }
-  } else {
-    return Status::InvalidArgument("external file version is not supported");
```
Comment (IslamAbdelRahman) [inline db/external_sst_file_ingestion_job.cc]: we should keep this as well (let's say we are ingesting V3 or V4, we should fail)

Comment (shuzhang1989) [inline db/external_sst_file_ingestion_job.cc]: okay

File: db/external_sst_file_ingestion_job.cc (hunk 1 above)
Comment (shuzhang1989) [inline db/external_sst_file_ingestion_job.cc]: okay

**Comment by shuzhang1989** [comment]
@IslamAbdelRahman updated

File: db/external_sst_file_ingestion_job.cc [hunk 3]
```diff
@@ -306,7 +307,13 @@ Status ExternalSstFileIngestionJob::GetIngestedFileInfo(
     if (file_to_ingest->global_seqno_offset == 0) {
       return Status::Corruption("Was not able to find file global seqno field");
     }
-  } else {
+  } else if (file_to_ingest->version == 1 &&
+        (ingestion_options_.allow_blocking_flush == true ||
+         ingestion_options_.allow_global_seqno == true)) {
+    return Status::InvalidArgument(
```
Comment (IslamAbdelRahman) [inline db/external_sst_file_ingestion_job.cc]: let's add an assert like this
```
// SST file V1 should not have global seqno field
assert(seqno_iter == uprops.end())
```

File: db/external_sst_file_ingestion_job.cc [hunk 4]
```diff
@@ -306,7 +307,13 @@ Status ExternalSstFileIngestionJob::GetIngestedFileInfo(
...
+            "external file's version number is not correct");
```
Comment (IslamAbdelRahman) [inline db/external_sst_file_ingestion_job.cc]: Let's update the error message to be something like this
```
"External SST file V1 does not support global seqno"
```

**Pull request closed without merging**